peer/brontide.go

```diff
@@ -777,6 +785,75 @@ func (p *Brontide) Start() error {
 	return nil
 }
 
+// peerStorageWriter coordinates persisting peer's backup data by delaying its
+// storage from its time of receipt to its time of persistence by the
+// duration specified in the `backupUpdateInterval`.
+func (p *Brontide) peerStorageWriter() {
+	defer p.wg.Done()
+
+	var data []byte
+
+Loop:
+	p.bMtx.L.Lock()
+	for {
+		p.bMtx.Wait()
+
+		// Store the data in a different variable as we are about to
+		// exit lock.
+		data = p.backupData
+		p.bMtx.L.Unlock()
+
+		// Check if we are to exit, now that we are awake.
+		select {
+		case <-p.quit:
+			if data == nil {
+				return
+			}
+
+			// Store the data immediately and exit.
+			err := p.cfg.PeerDataStore.Store(data)
+			if err != nil {
+				peerLog.Warnf("Failed to store peer "+
+					"backup data: %v", err)
+			}
+
+			return
```

Comment:
As discussed in the main PR thread this is not the approach we want to take. You shouldn't need a separate peerStorageWriter thread. It is fine to persist it in the main handler thread, or fork a single goroutine for overwriting the data so it doesn't block the main readHandler thread.

This will also alleviate the need to use `Cond`s which are notoriously difficult to use correctly.

Reply:
Do you mean in the `Store` method?